api/repositories/shared.go

```diff
@@ -59,3 +67,73 @@ func getLabelOrAnnotation(mapObj map[string]string, key string) string {
 	}
 	return mapObj[key]
 }
+
+func GetNamespacedResourceNamespace(ctx context.Context, privilegedClient client.Client, userClientFactory UserK8sClientFactory, authInfo authorization.Info, resourceGUID string, resourceList client.ObjectList) (string, error) {
+	err := privilegedClient.List(ctx, resourceList, client.MatchingFields{"metadata.name": resourceGUID})
+	if err != nil { // untested
+		return "", fmt.Errorf("privileged list failed: %w", err)
+	}
+
+	switch t := resourceList.(type) {
+	case *workloadsv1alpha1.CFAppList:
+		err := CheckListLength(len(t.Items), "app")
```

Comment:
The resource type could probably be passed as an argument, rather than the utility figuring it out by itself (repositories are pretty much aware of the resource types they are working with)